src/browser_action.js

```diff
@@ -157,16 +165,70 @@ browseraction.installButtonClickHandlers_ = function() {
 
 /** @private */
 browseraction.installKeydownHandlers_ = function() {
+  // Add new event to calendar on pressing `Ctrl + Enter`
   $('#quick-add-event-title').on('keydown', function(e) {
     // Check for Windows and Mac keyboards for event on Ctrl + Enter
-    if ((e.ctrlKey || e.metaKey) && (e.keyCode == 13 || e.keyCode == 10) &&
+    if ((e.ctrlKey || e.metaKey) &&
+        (e.keyCode == browseraction.KEY_CODE_CR || e.keyCode == browseraction.KEY_CODE_LF) &&
         $('#quick-add-event-title').val() !== '') {
       // Ctrl-Enter pressed
       browseraction.addNewEventIntoCalendar_();
     }
+
+    // Close quick add box, if empty, on `Esc`
+    if (e.keyCode == browseraction.KEY_CODE_ESCAPE) {
+      // Prevent popup from closing if quick-add-box is open and has unsaved input
+      e.stopPropagation();
+      e.preventDefault();
+
+      // Close quick add box if empty
+      if ($('#quick-add-event-title').val() === '') {
+        browseraction.toggleQuickAddBoxVisibility_(false);
+      }
+    }
+  });
+
+  // Open quick-add-box on pressing `a`
+  $(document).on('keypress', function(e) {
+    // Do nothing if in an input element
+    if ($(e.target).is('input, textarea, select')) {
+      return;
+    }
+
+    // Open quick add form on `a`
+    if (e.key.toLowerCase() === browseraction.SHORTCUT_OPEN_QUICK_ADD) {
+      e.stopPropagation();
+      e.preventDefault();
+      browseraction.toggleQuickAddBoxVisibility_(true);
+    }
   });
 };
 
+
+/** @private */
+browseraction.toggleQuickAddBoxVisibility_ = function(shouldShow) {
+  _gaq.push(['_trackEvent', 'Quick Add', 'Toggled']);
+
+  // If method is called from click-handler on FAB,
```

Comment:
Ah, this pattern would be better executed by attaching a different method to the click handler, and calling this method from it.